packages/apps/plugins/plugin-native/src/NativePlugin.tsx

```diff
@@ -81,6 +92,16 @@ export const initializeNativeApp = async () => {
     }
   });
 
+  window.addEventListener('applicationurl', (event: any) => {
+    const slug = event.url.host;
```

Comment:
- perhaps more compact and localizes logic below:
- also, please add a comment/link reference for applicationurl
- does it make sense to validate the url here or let the ACTIVATE function do that?

```tsx
window.addEventListener('applicationurl', ({ url }: any) => {
  if (!url.host.match(/../)) {
    // TODO(xxx): Better way to handle error than alert?
    alert();
    return;
  }

  void intentPlugin?.provides.intent.dispatch({
    action: LayoutAction.ACTIVATE,
    data: { id },
  });
});
```

Reply:
Because there is already an error message displayed by the app when the URL is not supported, I am going to remove the validity check here and always fire LayoutAction.ACTIVATE

Reply:
@mjamesderocher is this resolved? I still see the check below?

Reply:
@wittjosiah Yes, I had changed the code to be simply:

```// applicationurl is a custom event fired by the Socket Supply Runtime:
  // https://github.com/socketsupply/socket/blob/ef7fb5559876e41062d5896aafb7b79989fc96e5/api/internal/events.js#L6
  window.addEventListener('applicationurl', ({ url }: any) => {
    void intentPlugin?.provides.intent.dispatch({
      action: LayoutAction.ACTIVATE,
      data: { id: url.host },
    });
  });```
  
Am I missing something else that needs to be changed?

Reply:
I just thought you were saying we didn't need to validate here because ACTIVATE would do that, but seems like the code here is still checking `match` on the `slug` before navigating?

Reply:
@wittjosiah can you show me the lines where you are seeing a check for match? https://github.com/dxos/dxos/blob/7607cbcf2e8002d666fd3142669de5222630afd6/packages/apps/plugins/plugin-native/src/NativePlugin.tsx

Reply:
Ah my apologies I think I must've been looking at an outdate diff LGTM 👍